### ISSUE TITLE
Sentry Logging Time Issue

### ISSUE THREAD

**Issue by Vladelis** [getsentry/sentry-laravel#142]
Since Laravel 5.6 there is an logging config file thus making it easier to get the application logs onto sentry. The issue I kept running into is that the log displays its time with the default formatter and each log with the same message creates a new issue in sentry.
I made a solution by creating a CustomFormatter class and tapped it into the sentry channel to change the formatter into one that truncates the time from the message.
Is there a cleaner and better way of solving this? Seems like a lot of ugly code for some basic functionality.
```php
<?php

namespace Certificates\Logging;

use Monolog\Formatter\LineFormatter;

class CustomFormatter
{
    /**
     * @param  \Illuminate\Log\Logger  $logger
     * @return void
     */
    public function __invoke($logger)
    {
        foreach ($logger->getHandlers() as $handler) {
            $handler->setFormatter(new LineFormatter("%channel%.%level_name%: %message% %context% %extra%\n",null, false,true));
        }
    }
}
```

```php
        'sentry' => [
            'driver' => 'sentry',
            'tap' => [Certificates\Logging\CustomFormatter::class],
        ],
```

**Comment by stayallive** [comment]
This is not so nice, but I am not sure... probably Laravel is overwriting the line formatter since the format in the RavenHandler does not have any of that info.

See: https://github.com/Seldaek/monolog/blob/master/src/Monolog/Handler/RavenHandler.php

Open to suggestions on how to improve because dates in this format is not so good.

Maybe we should set the RavenHandler format explicitly when registering the log channel.

**Comment by Vladelis** [comment]
Yeah I think setting the default handler's formatter upon registration to one that does not display a different message for the same error (with time excluded), so that all of the same errors would fall under the same issue in Sentry would be a great solution.
Also I think the solution that the monolog driver has for logging without the use of the tap array is quite good. So maybe leave the option of changing the formatter or the handler log on the sentry driver in a similar way. 
Copied an example from the Laravel's documentation.
```PHP
'browser' => [
    'driver' => 'monolog',
    'handler' => Monolog\Handler\BrowserConsoleHandler::class,
    'formatter' => Monolog\Formatter\HtmlFormatter::class,
    'formatter_with' => [
        'dateFormat' => 'Y-m-d',
    ],
],
``` 
Thanks a lot for the effort

**Comment by dmyers** [comment]
I did a bit more research and it seems like the readme should be updated to recommend using a new SentryFormatter for the logs and possibly to add that formatter to this repo to make a good sensible default for people... I found this in the Monolog repo that is probably outdated at this point as it's named RavenHandler, but all it does is use the Monoglog\LineFormatter and change the format.

https://github.com/Seldaek/monolog/blob/master/src/Monolog/Handler/RavenHandler.php#L209

**Comment by dmyers** [comment]
I was able to get this to work now, but it might be beneficial to use this by default for others to have a good out-of-box setup.

```php
<?php
'sentry' => [
             'driver' => 'sentry',
             'level'  => 'error',
                  'formatter' => Monolog\Formatter\LineFormatter::class,
                         'formatter_with' => [
                               'format' => '%message%',
                   ],
         ],

**Comment by Vladelis** [comment]
@dmyers Yeah I think they patched it not long ago so your solution is what it should look like. I had to use a custom formatter beforehand which still worked, but was very unclean in my eyes.

**Comment by stayallive** [comment]
Happy to accept a PR to make this the default 👍

**Comment by sweet-greg** [comment]
Thank you for the fix @dmyers it helped bring down the duplicates quite a lot.
Yet, I see a lot of duplicates still, mostly initiated from Guzzle, I think it's because the error message is including the URL, so it is always a different message that is logged.
I found this [article](https://docs.sentry.io/data-management/rollups) explaining that grouping is done on message as a fallback. Would it be reasonably possible to add the stacktrace so grouping might become more robust? Or maybe I'm missing something obviously simple

**Comment by dmyers** [comment]
@sweet-greg Glad I could help. Yeah I have that same issue with Guzzle still I thought I had it fixed by creating a custom Exception inside `app/Exceptions/HttpClientException` that extends `\Exception` and I catch Guzzle's `RequestException` and rethrow it with my custom exception like this:

```php
try {
	// ...
}
catch (\GuzzleHttp\Exception\RequestException $e) {
	throw new HttpClientException(sprintf('Error with API call (%s)', $response->getReasonPhrase()), $response->getStatusCode());
}
```

Yeah I think stacktrace deduping would be really nice too, maybe someone from Sentry could comment on that idea or maybe a new issue.

**Comment by stayallive** [comment]
So this took a bit... but while working on some other log channel improvements I think we finally fixed this by default: #142.

We now just report the `%message%` without all the date, time and log level cruft now.
